csharp/ql/lib/semmle/code/csharp/dataflow/FlowSummary.qll

```diff
@@ -129,3 +152,17 @@ private class SummarizedCallableDefaultClearsContent extends Impl::Public::Summa
 }
 
 class RequiredSummaryComponentStack = Impl::Public::RequiredSummaryComponentStack;
+
+private class RecordConstructorFlowRequiredSummaryComponentStack extends RequiredSummaryComponentStack {
+  private SummaryComponent head;
+
+  RecordConstructorFlowRequiredSummaryComponentStack() {
+    exists(Property p |
+      recordConstructorFlow(_, _, p) and
+      head = SummaryComponent::property(p) and
+      this = SummaryComponentStack::singleton(SummaryComponent::return())
```

Comment:
I think this can be simplified to `this = SummaryComponentStack::return()`

Reply:
Yes, that is a good point! To be honest I just moved this code from the LibraryTypeDataFlow without looking that much into it 😸